pkg/cassandra/datacenter.go

```diff
@@ -450,25 +449,15 @@ func ValidateDatacenterConfig(dcConfig *DatacenterConfig) error {
 	return nil
 }
 
-func AddContainersToPodTemplateSpec(dcConfig *DatacenterConfig, containers ...corev1.Container) error {
-	if dcConfig.PodTemplateSpec == nil {
-		return errors.New("PodTemplateSpec was nil, cannot add containers")
-	} else {
-		dcConfig.PodTemplateSpec.Spec.Containers = append(dcConfig.PodTemplateSpec.Spec.Containers, containers...)
-		return nil
-	}
+func AddContainersToPodTemplateSpec(dcConfig *DatacenterConfig, containers ...corev1.Container) {
+	dcConfig.PodTemplateSpec.Spec.Containers = append(dcConfig.PodTemplateSpec.Spec.Containers, containers...)
 }
 
-func AddInitContainersToPodTemplateSpec(dcConfig *DatacenterConfig, initContainers ...corev1.Container) error {
-	if dcConfig.PodTemplateSpec == nil {
-		return errors.New("PodTemplateSpec was nil, cannot add init containers")
-	} else {
-		dcConfig.PodTemplateSpec.Spec.InitContainers = append(dcConfig.PodTemplateSpec.Spec.InitContainers, initContainers...)
-		position, found := FindInitContainer(dcConfig.PodTemplateSpec, reconciliation.ServerConfigContainerName)
-		if found && (dcConfig.PodTemplateSpec.Spec.InitContainers[position].Resources.Limits != nil || dcConfig.PodTemplateSpec.Spec.InitContainers[position].Resources.Requests != nil) {
-			dcConfig.ConfigBuilderResources = &dcConfig.PodTemplateSpec.Spec.InitContainers[position].Resources
-		}
-		return nil
+func AddInitContainersToPodTemplateSpec(dcConfig *DatacenterConfig, initContainers ...corev1.Container) {
+	dcConfig.PodTemplateSpec.Spec.InitContainers = append(dcConfig.PodTemplateSpec.Spec.InitContainers, initContainers...)
+	position, found := FindInitContainer(&dcConfig.PodTemplateSpec, reconciliation.ServerConfigContainerName)
```

Comment:
BTW I don't think this is the right place to inject config builder resources, this imho should be done in `NewDatacenter`.

Reply:
Went ahead and did the change.